Failed to publish gherkin/perl 21.0.0

The release of gherkin/perl 21.0.0 failed:

```
All tests successful.
Files=2, Tests=8,  3 wallclock secs ( 0.03 usr  0.02 sys +  0.37 cusr  0.16 csys =  0.58 CPU)
Result: PASS
cp ../CHANGELOG.md CHANGELOG.md
cpanm --notest --local-lib ./perl5 --installdeps --with-develop .
--> Working on .
Configuring /app/gherkin/perl ... OK
<== Installed dependencies for .. Finishing.
cpanm --notest --local-lib ./perl5 'Dist::Zilla'
--> Working on Dist::Zilla
Fetching http://www.cpan.org/authors/id/R/RJ/RJBS/Dist-Zilla-6.024.tar.gz ... OK
Configuring Dist-Zilla-6.024 ... OK
==> Found dependencies: Software::License
--> Working on Software::License
Fetching http://www.cpan.org/authors/id/R/RJ/RJBS/Software-License-0.104001.tar.gz ... OK
Configuring Software-License-0.104001 ... OK
Building Software-License-0.104001 ... OK
Successfully installed Software-License-0.104001 (upgraded from 0.103014)
Building Dist-Zilla-6.024 ... OK
Successfully installed Dist-Zilla-6.024 (upgraded from 6.017)
2 distributions installed
PERL5LIB=./perl5/lib/perl5 PATH=$PATH:./perl5/bin dzil authordeps --missing | cpanm --notest --local-lib ./perl5
--> Working on Dist::Zilla::Plugin::Git::GatherDir
Fetching http://www.cpan.org/authors/id/E/ET/ETHER/Dist-Zilla-Plugin-Git-2.048.tar.gz ... OK
Configuring Dist-Zilla-Plugin-Git-2.048 ... OK
==> Found dependencies: MooseX::Has::Sugar, Types::Path::Tiny, Type::Utils, Version::Next, Git::Wrapper, Dist::Zilla::Role::GitConfig, File::chdir, Types::Standard
--> Working on MooseX::Has::Sugar
Fetching http://www.cpan.org/authors/id/K/KE/KENTNL/MooseX-Has-Sugar-1.000006.tar.gz ... OK
Configuring MooseX-Has-Sugar-1.000006 ... OK
Building MooseX-Has-Sugar-1.000006 ... OK
Successfully installed MooseX-Has-Sugar-1.000006
--> Working on Types::Path::Tiny
Fetching http://www.cpan.org/authors/id/D/DA/DAGOLDEN/Types-Path-Tiny-0.006.tar.gz ... OK
Configuring Types-Path-Tiny-0.006 ... OK
==> Found dependencies: Types::TypeTiny, Type::Utils, Types::Standard, Type::Library
--> Working on Types::TypeTiny
Fetching http://www.cpan.org/authors/id/T/TO/TOBYINK/Type-Tiny-1.012004.tar.gz ... OK
Configuring Type-Tiny-1.012004 ... OK
Building Type-Tiny-1.012004 ... OK
Successfully installed Type-Tiny-1.012004
Building Types-Path-Tiny-0.006 ... OK
Successfully installed Types-Path-Tiny-0.006
--> Working on Version::Next
Fetching http://www.cpan.org/authors/id/D/DA/DAGOLDEN/Version-Next-1.000.tar.gz ... OK
Configuring Version-Next-1.000 ... OK
Building Version-Next-1.000 ... OK
Successfully installed Version-Next-1.000
--> Working on Git::Wrapper
Fetching http://www.cpan.org/authors/id/G/GE/GENEHACK/Git-Wrapper-0.048.tar.gz ... OK
==> Found dependencies: Devel::CheckBin
--> Working on Devel::CheckBin
Fetching http://www.cpan.org/authors/id/T/TO/TOKUHIROM/Devel-CheckBin-0.04.tar.gz ... OK
Configuring Devel-CheckBin-0.04 ... OK
Building Devel-CheckBin-0.04 ... OK
Successfully installed Devel-CheckBin-0.04
Configuring Git-Wrapper-0.048 ... OK
==> Found dependencies: File::chdir, Sort::Versions
--> Working on File::chdir
Fetching http://www.cpan.org/authors/id/D/DA/DAGOLDEN/File-chdir-0.1010.tar.gz ... OK
Configuring File-chdir-0.1010 ... OK
Building File-chdir-0.1010 ... OK
Successfully installed File-chdir-0.1010
--> Working on Sort::Versions
Fetching http://www.cpan.org/authors/id/N/NE/NEILB/Sort-Versions-1.62.tar.gz ... OK
Configuring Sort-Versions-1.62 ... OK
Building Sort-Versions-1.62 ... OK
Successfully installed Sort-Versions-1.62
Building Git-Wrapper-0.048 ... OK
Successfully installed Git-Wrapper-0.048
--> Working on Dist::Zilla::Role::GitConfig
Fetching http://www.cpan.org/authors/id/B/BB/BBYRD/Dist-Zilla-Plugin-Config-Git-0.92.tar.gz ... OK
Configuring Dist-Zilla-Plugin-Config-Git-0.92 ... OK
Building Dist-Zilla-Plugin-Config-Git-0.92 ... OK
Successfully installed Dist-Zilla-Plugin-Config-Git-0.92
Building Dist-Zilla-Plugin-Git-2.048 ... OK
Successfully installed Dist-Zilla-Plugin-Git-2.048
--> Working on Dist::Zilla::Plugin::MetaProvides::Package
Fetching http://www.cpan.org/authors/id/K/KE/KENTNL/Dist-Zilla-Plugin-MetaProvides-Package-2.004003.tar.gz ... OK
Configuring Dist-Zilla-Plugin-MetaProvides-Package-2.004003 ... OK
==> Found dependencies: Dist::Zilla::Role::ModuleMetadata, Safe::Isa, Dist::Zilla::Role::MetaProvider::Provider, Dist::Zilla::Plugin::MetaProvides, Dist::Zilla::MetaProvides::ProvideRecord
--> Working on Dist::Zilla::Role::ModuleMetadata
Fetching http://www.cpan.org/authors/id/E/ET/ETHER/Dist-Zilla-Role-ModuleMetadata-0.006.tar.gz ... OK
Configuring Dist-Zilla-Role-ModuleMetadata-0.006 ... OK
Building Dist-Zilla-Role-ModuleMetadata-0.006 ... OK
Successfully installed Dist-Zilla-Role-ModuleMetadata-0.006
--> Working on Safe::Isa
Fetching http://www.cpan.org/authors/id/E/ET/ETHER/Safe-Isa-1.000010.tar.gz ... OK
Configuring Safe-Isa-1.000010 ... OK
Building Safe-Isa-1.000010 ... OK
Successfully installed Safe-Isa-1.000010
--> Working on Dist::Zilla::Role::MetaProvider::Provider
Fetching http://www.cpan.org/authors/id/K/KE/KENTNL/Dist-Zilla-Plugin-MetaProvides-2.002004.tar.gz ... OK
Configuring Dist-Zilla-Plugin-MetaProvides-2.002004 ... OK
==> Found dependencies: Hash::Merge::Simple
--> Working on Hash::Merge::Simple
Fetching http://www.cpan.org/authors/id/R/RO/ROKR/Hash-Merge-Simple-0.051.tar.gz ... OK
Configuring Hash-Merge-Simple-0.051 ... OK
==> Found dependencies: Test::Most
--> Working on Test::Most
Fetching http://www.cpan.org/authors/id/O/OV/OVID/Test-Most-0.37.tar.gz ... OK
Configuring Test-Most-0.37 ... OK
==> Found dependencies: Test::Warn
--> Working on Test::Warn
Fetching http://www.cpan.org/authors/id/B/BI/BIGJ/Test-Warn-0.36.tar.gz ... OK
Configuring Test-Warn-0.36 ... OK
Building Test-Warn-0.36 ... OK
Successfully installed Test-Warn-0.36
Building Test-Most-0.37 ... OK
Successfully installed Test-Most-0.37
Building Hash-Merge-Simple-0.051 ... OK
Successfully installed Hash-Merge-Simple-0.051
Building Dist-Zilla-Plugin-MetaProvides-2.002004 ... OK
Successfully installed Dist-Zilla-Plugin-MetaProvides-2.002004
Building Dist-Zilla-Plugin-MetaProvides-Package-2.004003 ... OK
Successfully installed Dist-Zilla-Plugin-MetaProvides-Package-2.004003
Dist::Zilla::Plugin::Prereqs::FromCPANfile is up to date. (0.08)
17 distributions installed
PERL5LIB=./perl5/lib/perl5 PATH=$PATH:./perl5/bin dzil clean
Can't locate Module/CPANfile.pm in @INC (you may need to install the Module::CPANfile module) (@INC contains: ./perl5/lib/perl5/x86_64-linux-gnu-thread-multi ./perl5/lib/perl5 /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.30.0 /usr/local/share/perl/5.30.0 /usr/lib/x86_64-linux-gnu/perl5/5.30 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.30 /usr/share/perl/5.30 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base /app/gherkin/perl) at perl5/lib/perl5/Dist/Zilla/Plugin/Prereqs/FromCPANfile.pm line 6, <GEN0> line 28.
BEGIN failed--compilation aborted at perl5/lib/perl5/Dist/Zilla/Plugin/Prereqs/FromCPANfile.pm line 6, <GEN0> line 28.
Compilation failed in require at perl5/lib/perl5/Module/Runtime.pm line 314, <GEN0> line 28. at perl5/lib/perl5/Config/MVP/Assembler.pm line 144.
make[1]: *** [default.mk:55: predistribution] Error 2
make[1]: Leaving directory '/app/gherkin/perl'
make: *** [default.mk:54: publish-perl] Error 2
```

I tried to publish it again after it failed, but it gave the same error:

```
# running in the docker container, checked out the gherkin/v21.0.0 tag
cd gherkin/perl
NEW_VERSION=21.0.0 make publish
```

--- a/.templates/perl/default.mk
+++ b/.templates/perl/default.mk
@@ -46,7 +46,7 @@ endif
 	cpanm --notest --local-lib ./perl5 --installdeps .
 	touch $@
 
-predistribution: test CHANGELOG.md
+predistribution: dist-clean test CHANGELOG.md
 # --notest to keep the number of dependencies low: it doesn't install the
 # testing dependencies of the dependencies.
 	cpanm --notest --local-lib ./perl5 --installdeps --with-develop .
@@ -62,3 +62,8 @@ pre-release: update-version
 
 post-release:
 .PHONY: post-release
+
+dist-clean: clean
+	rm -rf ./perl5
+
+.PHONY: dist-clean

--- a/gherkin/perl/Makefile
+++ b/gherkin/perl/Makefile
@@ -8,14 +8,12 @@ ASTS     = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.ast.ndjso
 PICKLES  = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.pickles.ndjson,$(GOOD_FEATURE_FILES))
 ERRORS   = $(patsubst testdata/%.feature,acceptance/testdata/%.feature.errors.ndjson,$(BAD_FEATURE_FILES))
 
-PERL_FILES = $(shell find . -name "*.pm")
-
 .DELETE_ON_ERROR:
 
 test: .built $(TOKENS) $(ASTS) $(PICKLES)
 	PERL5LIB=./perl5/lib/perl5 prove -l
 
-.built: .cpanfile_dependencies lib/Gherkin/Generated/Parser.pm lib/Gherkin/Generated/Languages.pm $(PERL_FILES) bin/gherkin-generate-tokens LICENSE.txt
+.built: .cpanfile_dependencies lib/Gherkin/Generated/Parser.pm lib/Gherkin/Generated/Languages.pm bin/gherkin-generate-tokens LICENSE.txt
 	@$(MAKE) --no-print-directory show-version-info
 	# add Perl-level unit tests
 	touch $@

--- a/gherkin/perl/default.mk
+++ b/gherkin/perl/default.mk
@@ -46,7 +46,7 @@ endif
 	cpanm --notest --local-lib ./perl5 --installdeps .
 	touch $@
 
-predistribution: test CHANGELOG.md
+predistribution: dist-clean test CHANGELOG.md
 # --notest to keep the number of dependencies low: it doesn't install the
 # testing dependencies of the dependencies.
 	cpanm --notest --local-lib ./perl5 --installdeps --with-develop .
@@ -62,3 +62,8 @@ pre-release: update-version
 
 post-release:
 .PHONY: post-release
+
+dist-clean: clean
+	rm -rf ./perl5
+
+.PHONY: dist-clean

--- a/messages/perl/default.mk
+++ b/messages/perl/default.mk
@@ -46,7 +46,7 @@ endif
 	cpanm --notest --local-lib ./perl5 --installdeps .
 	touch $@
 
-predistribution: test CHANGELOG.md
+predistribution: dist-clean test CHANGELOG.md
 # --notest to keep the number of dependencies low: it doesn't install the
 # testing dependencies of the dependencies.
 	cpanm --notest --local-lib ./perl5 --installdeps --with-develop .
@@ -62,3 +62,8 @@ pre-release: update-version
 
 post-release:
 .PHONY: post-release
+
+dist-clean: clean
+	rm -rf ./perl5
+
+.PHONY: dist-clean

